com.unity.ml-agents/Runtime/Academy.cs

```diff
@@ -205,11 +205,21 @@ public void DisableAutomaticStepping(bool destroyImmediate = false)
         }
 
         /// <summary>
-        /// Returns whether or not the Academy is automatically stepped during the FixedUpdate phase.
+        /// Determines whether or not the Academy is automatically stepped during the FixedUpdate phase.
         /// </summary>
-        public bool IsAutomaticSteppingEnabled
+        public bool AutomaticSteppingEnabled
         {
             get { return m_FixedUpdateStepper != null; }
+            set {
+                if (value)
+                {
+                    EnableAutomaticStepping();
+                }
+                else
+                {
+                    DisableAutomaticStepping();
```

Comment:
Why is  DisableAutomaticStepping still public? Seems like there are now 2 ways to disable stepping 
```
AutomaticSteppingEnabled = false;
```
and
```
DisableAutomaticStepping()
```

If it is to leave the option to destroy immediate or not, why do we need to be able to do both ?

Reply:
From the comments above:
>  I assume Object.Destroy() is generally preferable, but unit tests break if we do that instead of Object.DestroyImmediate()

Open to other suggestions on how to resolve this.

Reply:
What is wrong having DestroyImmediate be the only option?

Reply:
Or we could have this in the test : 
```csharp
[Test]
        public void TestAcademyAutostep()
        {
            var aca = Academy.Instance;
            Assert.IsTrue(aca.IsAutomaticSteppingEnabled);
            UnityEngine.TestTools.LogAssert.Expect(LogType.Error, "Destroy may not be called from edit mode! Use DestroyImmediate instead.\nAlso think twice if you really want to destroy something in edit mode. Since this will destroy objects permanently.");
            aca.DisableAutomaticStepping(false);
            Assert.IsFalse(aca.IsAutomaticSteppingEnabled);
            aca.EnableAutomaticStepping();
            Assert.IsTrue(aca.IsAutomaticSteppingEnabled);
        }
```

Reply:
DestroyImmediate is strongly recommended against since it tries to deallocate memory immediately instead of batching it together after a normal Destroy call.

Reply:
Should we make `DisableAutomaticStepping` internal (This way we can test it without exposing it?)

Reply:
DisableAutomaticStepping is now private; I use Application.isEditor to switch between Destroy and DestoryImmediate.